### ISSUE TITLE
[Bug]: When using a screen-reader, the user is unable to 'tab' into the dismiss button

### ISSUE THREAD

**Issue by syphe** [microsoft/fluentui#30682]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.22631
    CPU: (16) x64 AMD EPYC 7763 64-Core Processor
    Memory: 32.60 GB / 63.95 GB
  Browsers:
    Edge: Chromium (121.0.2277.128)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://codesandbox.io/p/sandbox/fluentui-searchbox-a11y-clear-button-jhgwvz?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clt8p2m6100063j6fbze6elun%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clt8p2m6100023j6f8owtqnrj%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clt8p2m6100033j6frpyc1jkw%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clt8p2m6100053j6f7p2i31i9%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clt8p2m6100023j6f8owtqnrj%2522%253A%257B%2522id%2522%253A%2522clt8p2m6100023j6f8owtqnrj%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clt8p2m6100053j6f7p2i31i9%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clt8p2m6100043j6fc6tvy8xs%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clt8p2m6100053j6f7p2i31i9%2522%252C%2522activeTabId%2522%253A%2522clt8p2m6100043j6fc6tvy8xs%2522%257D%252C%2522clt8p2m6100033j6frpyc1jkw%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clt8p2m6100033j6frpyc1jkw%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D

## Actual Behavior
When user enters text into the searchbox, and then presses the tab key, the focus is moved out of the search box component

## Expected Behavior
When user enters text into the searchbox, and then presses the tab key, the dismiss button should be focused, so that users with screen readers are able to clear the search box.

### Requested priority

Normal

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by syphe** [comment]
Have created a PR to address this issue: https://github.com/microsoft/fluentui/pull/30677

**Comment by miroslavstastny** [comment]
I think that to dismiss the content of the search box by keyboard, user is supposed to press Esc key.
cc @smhigley

**Comment by syphe** [comment]
> I think that to dismiss the content of the search box by keyboard, user is supposed to press Esc key. cc @smhigley

I think for accessibility reasons, the user should be able to navigate to the dismiss button using tabs. I'm not sure if pressing escape is obvious to a user with a screen reader.

**Comment by smhigley** [comment]
Specifying screen readers is a bit of a red herring -- this is a keyboard behavior, independent of screen reader usage.

This is also by design, as @miroslavstastny mentioned. The clear button is a pointer-specific affordance (that also works for voice control, touch, and touch screen readers) to clear the text easily. Keyboard users can select-all + backspace to clear text.

We do not expose the clear button in the tab order because it would add an extra tab stop per input for no meaningful benefit. We chose this behavior as a consensus with other a11y SMEs in the company, including screen reader users. This also follows the behavior of the native browser-rendered clear button, which is not inserted into the tab order.
